doc/vlan/switchport-mode-support/Switchport Mode and VLAN CLI Enhancement.md

```diff
@@ -0,0 +1,608 @@
+
+**4.** Improving network scalability: Trunking also improves network scalability. By allowing multiple VLANs to be transmitted over a single port, you can increase the number of devices that can be connected to the network without needing additional switches.
+
+Overall, the switch port mode trunk is a useful feature for connecting switches and other network devices together, improving network performance, and providing redundancy. By enabling trunking, users can simplify network topology, reduce congestion, and increase network scalability.
+
+## Architecture Design
+
+The overall SONiC architecture will remain the same and no new sub-modules will be introduced. Changes are made only in the CLI container and Config_DB.
+
+![Architecture](https://user-images.githubusercontent.com/61490193/217232727-811b98a8-ce9c-4227-a94a-1c83a4825a62.png)
+
+
+## High-level Design
+
+In this section we will explain sequence diagrams for the implemented features.
+
+### State Transition Diagram of Switchport modes for PORT or PORT CHANNEL
+
+The following state transition depicts the behavior of Switchport  modes access and trunk for adding port modes on a Port or PortChannel. Switching between these modes is also depicted here. Default port mode is “routed”.
+
+![State-Transition-Diagram](https://user-images.githubusercontent.com/61490193/223195784-05cc4e98-8883-4911-bca3-fded6a87d202.png)
+
+__*Figure 2: State Transition Diagram SwitchPort mode*__
+
+
+### Sequence Diagram for Adding Multiple VLANs
+ 
+
+![Add-Vlan](https://user-images.githubusercontent.com/61490193/223186173-b1eecf22-eef0-4ba1-86d1-8675a7b00f31.png)
+
+__*Figure 3: Sequence Diagram for adding Multiple VLANs*__
+
+**1.** The SONiC CLI will call the multiple vlan list parser which will parse multiple vlan(s) in a list and return.
+
+**2.** Check if the VLAN is in the range (2-4094) and whether the vid already exists or not.
+
+**3.** Check if vlan id already exist it will show an error message and truncate. Examples for such cases have been mentioned in the examples section below.
+
+**4.** After checking the vlan list in the range and vlan doesn’t already exist.
+
+**5.** CLI will add the VLAN from the list in the VLAN table.
+
+__*NOTE: This works same for Adding Multiple VLANs on a PortChannel*__
+
+
+### Seqence Diagram for Deleting Multiple VLANs
+
+
+![Delete-Vlan](https://user-images.githubusercontent.com/61490193/223189728-8545972f-c726-4d6a-a66c-984617e825c2.png)
+
+
+__*Figure 4: Sequence Diagram for deleting Multiple VLANs*__
+
+**1.** The SONiC CLI will call the multiple vlan list parser which will parse multiple vlan(s) in a list.
+
+**2.** Check if the VLAN is in the range .
+
+**3.** Check if vlan exists or not. If vlan does not exist it will show an error message and truncate. Examples for such cases have been mentioned in the examples section as well.
+
```

Comment:
If the VLAN still has member ports, you should not allow delete of the VLAN right?  Or do you implicitly remove the port membership to the VLAN being deleted?

Reply:
We did not change the current functionality of SONiC with regards to VLAN and Member port. If the VLAN has member assigned to it and user tries to delete the VLAN, CLI will give an error. Please refer to examples section, we have already included example of this behavior.